CRM/Utils/Recent.php

```diff
@@ -188,4 +204,51 @@ public static function delContact($id) {
     $session->set(self::STORE_NAME, self::$_recent);
   }
 
+  /**
+   * Check if a provider is allowed to add stuff.
+   * If correspondig setting is empty, all are allowed
+   *
+   * @param string $providerName
+   */
+  public static function isProviderEnabled($providerName) {
```

Comment:
It's public because providers may check it before adding items to the stack. But you're right at the moment this isn't used anymore since I utilized the $type param.